AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift

```diff
@@ -0,0 +1,432 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+
+import AWSPluginsCore
+import Combine
+import AWSAPIPlugin
+
+@testable import Amplify
+@testable import AWSDataStorePlugin
+@testable import DataStoreHostApp
+
+final class DataStoreStressTests: XCTestCase {
+
+    static let amplifyConfigurationFile = "testconfiguration/AWSDataStoreStressTests-amplifyconfiguration"
+    let concurrencyLimit = 50
+
+    let networkTimeout = TimeInterval(180)
+    
+    struct TestModelRegistration: AmplifyModelRegistration {
+        func registerModels(registry: ModelRegistry.Type) {
+            registry.register(modelType: Post.self)
+            registry.register(modelType: Comment.self)
+        }
+
+        let version: String = "1"
+    }
+
+    func setUp(withModels models: AmplifyModelRegistration, logLevel: LogLevel = .error) async {
+        continueAfterFailure = false
+        await Amplify.reset()
+        Amplify.Logging.logLevel = logLevel
+
+        do {
+            try Amplify.add(plugin: AWSAPIPlugin(modelRegistration: models))
+            try Amplify.add(plugin: AWSDataStorePlugin(modelRegistration: models))
```

Comment:
After reviewing the tests being added, I don't see any tests that stress the performance of the initial DataStore sync, so it will be better for the latency of a test by setting the max sync count to 100. Default is 10000 which will be the normal amount synced down after a few CI test runs against this backend.